src/vivarium_gates_mncnh/components/intrapartum.py

```diff
@@ -44,22 +50,30 @@ def on_time_step(self, event: Event) -> None:
             return
 
         pop = self.population_view.get(event.index)
-        facility_type_mapper = {
-            DELIVERY_FACILITY_TYPES.BEmONC: NO_CPAP_RISK.P_CPAP_BEmONC,
-            DELIVERY_FACILITY_TYPES.CEmONC: NO_CPAP_RISK.P_CPAP_CEmONC,
-        }
 
-        for facility_type in [
-            DELIVERY_FACILITY_TYPES.BEmONC,
-            DELIVERY_FACILITY_TYPES.CEmONC,
-        ]:
+        for facility_type in self.delivery_facility_access_probabilities:
             facility_idx = pop.index[pop[COLUMNS.DELIVERY_FACILITY_TYPE] == facility_type]
-            cpap_access_probability = CPAP_ACCESS_PROBABILITIES[self.location][
-                facility_type_mapper[facility_type]
-            ]
             cpap_access_idx = self.randomness.filter_for_probability(
-                facility_idx, cpap_access_probability, f"cpap_access_{facility_type}"
+                facility_idx,
+                self.coverage_values[facility_type],
+                f"cpap_access_{facility_type}",
             )
             pop.loc[cpap_access_idx, COLUMNS.CPAP_AVAILABLE] = True
 
         self.population_view.update(pop)
+
+    def get_coverage_values(self) -> dict[str, float]:
```

Comment:
If you pass `builder` to this method, you can just access the artifact directly rather than going via the intermediary of `self.delivery_facility_access_probabilities`

When you make your next intervention component (for antibiotics), you can think about how to generalize this method.